tests/external_botocore/test_bedrock_chat_completion.py

```diff
@@ -416,5 +440,366 @@ def _test():
+        rollup_metrics=[("Llm/completion/Bedrock/invoke_model", 1)],
+        custom_metrics=[
+            ("Supportability/Python/ML/Bedrock/%s" % BOTOCORE_VERSION, 1),
+        ],
+        background_task=True,
+    )
+    @background_task(name="test_bedrock_chat_completion")
+    def _test():
+        model = "amazon.titan-text-express-v1"
+        body = (chat_completion_payload_templates[model] % ("Malformed Body", 0.7, 100)).encode("utf-8")
+        set_trace_info()
+        add_custom_attribute("llm.conversation_id", "my-awesome-id")
+        add_custom_attribute("llm.foo", "bar")
+        add_custom_attribute("non_llm_attr", "python-agent")
+
+        response = bedrock_server.invoke_model(
+            body=body,
+            modelId=model,
+            accept="application/json",
+            contentType="application/json",
+        )
+        assert response
+
+    _test()
+
+
+@reset_core_stats_engine()
+def test_bedrock_chat_completion_error_malformed_response_streaming_body(
+    bedrock_server,
+    set_trace_info,
+):
+    """
+    After a streaming call, the chunk returned by the server is valid, but contains a body with JSON that cannot be parsed.
+
+    * No actual error should be raised by the library or our code.
+    * An invalid payload is returned by the library to the user.
+    * No response can be parsed, meaning all response data should be missing.
+    """
+
+    @validate_custom_events(chat_completion_expected_malformed_response_streaming_body_events)
+    @validate_custom_event_count(count=2)
+    @validate_transaction_metrics(
+        name="test_bedrock_chat_completion",
+        scoped_metrics=[("Llm/completion/Bedrock/invoke_model_with_response_stream", 1)],
+        rollup_metrics=[("Llm/completion/Bedrock/invoke_model_with_response_stream", 1)],
+        custom_metrics=[
+            ("Supportability/Python/ML/Bedrock/%s" % BOTOCORE_VERSION, 1),
+        ],
+        background_task=True,
+    )
+    @background_task(name="test_bedrock_chat_completion")
+    def _test():
+        model = "amazon.titan-text-express-v1"
+        body = (chat_completion_payload_templates[model] % ("Malformed Streaming Body", 0.7, 100)).encode("utf-8")
+
+        set_trace_info()
+        add_custom_attribute("llm.conversation_id", "my-awesome-id")
+        add_custom_attribute("llm.foo", "bar")
+        add_custom_attribute("non_llm_attr", "python-agent")
+
+        response = bedrock_server.invoke_model_with_response_stream(
+            body=body,
+            modelId=model,
+            accept="application/json",
+            contentType="application/json",
+        )
+
+        chunks = list(response["body"])
+        assert chunks, "No response chunks returned"
+        for chunk in chunks:
+            with pytest.raises(json.decoder.JSONDecodeError):
+                json.loads(chunk["chunk"]["bytes"])
+
+    _test()
+
+
+_headers_error = botocore.eventstream.InvalidHeadersLength
```

Comment:
These are only used in one place. Let's move these to where they are used instead of abstracting them into a constant somewhere else in the test file. This is a really terrible pattern we seem to use all over our tests. It makes the code a lot harder to read when we do this. If it's used in multiple places-it makes sense to use this pattern but here, it's only used once so let's just put it where it's used-there's no reason to add a layer of abstraction.